qa/tasks/cephfs/test_misc.py

```diff
@@ -187,6 +190,181 @@ def test_dump_inode_hexademical(self):
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("Require FUSE client")
+
+        self.mount_a.umount_wait()
+
+        timeout = 30
+        self.set_conf('client', 'client caps wanted delay max', timeout)
+
+        self.mount_a.mount()
+        self.mount_a.wait_until_mounted()
+        mount_a_client_id = self.mount_a.get_global_id()
+
+        filename = "testfile"
+        path = os.path.join(self.mount_a.mountpoint, filename)
+        pyscript = dedent("""
+            import os
+            import sys
+
+            fd = os.open("{path}", os.O_RDWR | os.O_CREAT, 0O666)
+            os.write(fd, b'content')
+
+            print("write done")
+            sys.stdout.flush()
+
+            sys.stdin.readline()
+
+            os.write(fd, b'content')
+            """).format(path=path)
+
+        rproc = self.mount_a.client_remote.run(
+                    args=['sudo', 'python3', '-c', pyscript],
+                    wait=False, stdin=run.PIPE, stdout=run.PIPE)
+
+        rproc.stdout.readline()
+
+        ino = self.mount_a.path_to_ino(filename)
+
+        info = self.fs.mds_asok(['dump', 'inode', hex(ino)])
+        caps = filter(lambda c: c['client_id'] == mount_a_client_id, info['client_caps'] )[0]
+        log.info("caps = {}".format(caps))
+        self.assertNotEqual(caps['wanted'], '-')
+
+        time.sleep(timeout * 2)
+
+        info = self.fs.mds_asok(['dump', 'inode', hex(ino)])
+        caps = filter(lambda c: c['client_id'] == mount_a_client_id, info['client_caps'] )[0]
+        log.info("caps = {}".format(caps))
+        self.assertEqual(caps['wanted'], '-')
+
+        rproc.stdin.writelines(['continue\n'])
+        rproc.stdin.flush()
+
+        rproc.wait()
+        self.assertEqual(rproc.exitstatus, 0)
+
+class TestAsyncDirOps(CephFSTestCase):
+    CLIENTS_REQUIRED = 2
+
+    def _setup(self):
+        # enable async dirops on mount_a. mount_b is for checking the results.
+        if isinstance(self.mount_a, FuseMount):
+            self.mount_a.umount_wait()
+            self.mount_a.mount(mount_options=['--client_async_dirop_mask=3'])
+            self.mount_a.wait_until_mounted()
+            # prevent lookup request from changing target inode's lock state.
+            self.mount_b.umount_wait()
+            self.mount_b.mount(mount_options=['--client_debug_stat_cap_mask=0'])
+            self.mount_b.wait_until_mounted()
+        else:
+            self.mount_a.umount_wait()
+            try:
+                self.mount_a.mount(mount_options=['nowsync'])
+            except CommandFailedError:
+                self.mount_a.kill_cleanup()
+                self.skipTest("Async dirop is not implemented in current kernel")
+
+    def test_create(self):
+        self._setup()
+
+        self.mount_a.run_shell(['mkdir', 'testdir'])
+        self.mount_a.run_shell(['touch', 'testdir/file1'])
+
+        self.fs.mds_asok(["config", "set", "mds_early_reply", "0"])
```

Comment:
`self.config_set`

Reply:
(and everywhere else)

Reply:
we need to change config of running mds.  self.config_set does not work.

Reply:
Yes it should? You might be thinking of the old `set_conf`.

Reply:
this config is temporary. self.config_set() set the config in mon's config database